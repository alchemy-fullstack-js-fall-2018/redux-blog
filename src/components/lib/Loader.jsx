import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledImg = styled.img`
  object-fit: contain;
`;

const Input = ({ src }) => {

  return (
    <StyledImg
      src={src}
      alt={'loading'}
    />
  );
};

Input.propTypes = {
  src: PropTypes.string.isRequired
};

export default Input;
