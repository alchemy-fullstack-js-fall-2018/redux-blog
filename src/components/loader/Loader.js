import React, { Fragment } from 'react';




const Loader = () => {

  return (
    <Fragment>
      <h1>Loading</h1>
    </Fragment>
  );
};


export default Loader;
