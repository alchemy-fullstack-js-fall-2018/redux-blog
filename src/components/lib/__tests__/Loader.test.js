import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../Loader.jsx';

describe('<Loader />', () => {
  test('renders', () => {
    const src = 'https://media.giphy.com/media/l41lFw057lAJQMwg0/giphy.gif';
    const wrapper = shallow(
      <Loader
        src={src}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
