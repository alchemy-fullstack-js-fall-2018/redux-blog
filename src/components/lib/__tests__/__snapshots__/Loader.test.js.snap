// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders 1`] = `
<styled.img
  alt="loading"
  src="https://media.giphy.com/media/l41lFw057lAJQMwg0/giphy.gif"
/>
`;
